### DISCUSSION TITLE
Does a celery task support asyncio operations ?

### DISCUSSION THREAD

**Discussion by yichong96** [celery/celery#7176] [Q&A]
I Would like to know whether it is possible to have async/awaits while celery is running the task. I envision that while the worker process is awaiting for the network I/O to complete within the task, the cpu can possibly context switch to another process and then come back to the task once the "await" operation completes. 

I read that celery does have support for gevent task and the lightweight worker threads are suited for network I/O bound task. The task that I am running however is an image processing task in which I am sending images over a network to another server that will perform inference on it. Therefore the network I/O would not be as fast as the usual GET/POST requests. While gevent can provide concurrent sending of HTTP request, I will have to call a blocking function to wait for all responses.

Thank you for the help and advice !

**Comment by thedrow** [comment]
Celery does not support asyncio. There are multiple issues about it #6552 #6603.
I'm working on a new approach that seems promising.
